### ISSUE TITLE
/user endpoint issues - primarily 'debt'

### ISSUE THREAD

**Issue by rasmuserik** [DBCDK/serviceprovider#467]
For anonymous user, `dbcOpenPlatform.user()` yielded:

```json
{
  "id": "Wdg/d0ONAh8lOVP+Rp13MXc2MpkK0Bx4",
  "loans": [
    {
      "loanId": "NCIPMDAxOXw5OTAwfHw5OTAwLnx8",
      "dueDate": "2016-06-20T00:00:00+02:00",
      "title": "9900."
    },
    {
      "loanId": "NCIPMDAxOXwwMDAwMDMyNDg0fFJvc2VuLCBNaWNoYWVsfFNtdXR8fHw=",
      "dueDate": "2016-06-27T00:00:00+02:00",
      "title": "Smut",
      "author": "Rosen, Michael"
    }
  ],
  "orders": [
    {
      "title": "Turen går til Gran Canaria, Fuerteventura og Lanzarote",
      "orderId": "Hold:NCIPMDAxOXwyNy0wNS0yMDE2IDEwOjA3OjU4fExvdW1hbm4sIE9sZXxUdXJlbiBnJmFyaW5nO3IgdGlsIEdyYW4gQ2FuYXJpYSwgRnVlcnRldmVudHVyYSBvZyBMYW56YXJvdGV8fHwx",
      "orderDate": "2016-05-27T00:00:00+02:00",
      "status": "In process",
      "library": "DK-100451",
      "holdQueuePosition": "1",
      "author": "Loumann, Ole"
    },
    {
      "title": "Evas sidste nat",
      "orderId": "Hold:NCIPMDAxOXwyNy0wNS0yMDE2IDA5OjU4OjU0fEdlcmxvdywgSmVhbmV0dGUgJk9zbGFzaDticm98RXZhcyBzaWRzdGUgbmF0fHx8",
      "orderDate": "2016-05-27T00:00:00+02:00",
      "status": "In process",
      "library": "DK-100451",
      "author": "Gerlow, Jeanette Øbro"
    },
    {
      "title": "Det ·norske job",
      "orderId": "Hold:NCIPMDAxOXwyNy0wNS0yMDE2IDEwOjA0OjAyfEdlcmxvdywgSmVhbmV0dGUgJk9zbGFzaDticm98RGV0ICZtaWRkb3Q7bm9yc2tlIGpvYnx8fA==",
      "orderDate": "2016-05-27T00:00:00+02:00",
      "status": "In process",
      "library": "DK-100451",
      "author": "Gerlow, Jeanette Øbro"
    }
  ],
  "ddbcmsapi": "http://rest.filmstriben.dbc.inlead.dk/web/"
}
```
Where orders/loans should probably be empty for this particular user.

... and a bit later(now) it returns:

```js
{statusCode: 500, error: "TypeError: promise.then is not a function"}
```

... and later yet, anonymous logins does not work, (after switching to borchk).

**Comment by rasmuserik** [comment]
With a logged in user, it currently fails as:
```
{statusCode: 500, error: "TypeError: Cannot read property 'loanedItems' of undefined"}
```

**Comment by rasmuserik** [comment]
debt section of `/user` is missing.
(and there is also a bug in the spec, where debt is an object, and should be an array of objects).

**Comment by cboesgaard** [comment]
This issue is now on debt, other issues have been moved into other issues.

The current plan is to implement debt in a later version.

**Comment by rasmuserik** [comment]
Fixed with #515
